src/gd_png.c

```diff
@@ -414,6 +414,9 @@ BGD_DECLARE(gdImagePtr) gdImageCreateFromPngCtx (gdIOCtx * infile)
 		gd_error("gd-png color_type is unknown: %d\n", color_type);
 		goto error;
 	}
+	
+	/* interlacing handled by libpng */
+	png_set_interlace_handling(png_ptr);
```

Comment:
Given the function:

```C
png_set_interlace_handling(png_structrp png_ptr)
{
   png_debug(1, "in png_set_interlace handling");

   if (png_ptr != 0 && png_ptr->interlaced != 0)
   {
      png_ptr->transformations |= PNG_INTERLACE;
      return (7);
   }

   return (1);
}
```

Either we do not care about the image being interlaced or not at this stage, or we do and we should handle the return value.

If we don't, the could can be changed to:

```C
(void)png_set_interlace_handling(png_ptr);
```

Otherwise that function returns the number of passes afair.

If I am not mistaken, libpng can also be built without interlace support, meaning #ifdef PNG_READ_INTERLACING_SUPPORTED needs to be used here.